app/pages/WhenWantPetUntilPage.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pages
+
+import java.time.LocalDate
+
+import play.api.libs.json.JsPath
+
+case object WhenWantPetUntilPage extends QuestionPage[LocalDate] {
+
+  override def path: JsPath = JsPath \ toString
+
+  override def toString: String = "whenWantPetUntil"
```

Comment:
I think "When" because it's like "Given/When/Then" confused me a bit and I had to read stuff a few time to get it into my brain

I've forgotten what the journey is it essentially

RentalPeriodEndDate

Not saying change anything, just a "when" might confuse people